ioeffect/src/main/scala/scalaz/ioeffect/IO.scala

```diff
@@ -768,6 +777,24 @@ object IO extends IOInstances {
   final def require[E, A](error: E): IO[E, Maybe[A]] => IO[E, A] =
     (io: IO[E, Maybe[A]]) => io.flatMap(_.cata(IO.now[E, A](_), IO.fail[E, A](error)))
 
+  /**
+   * Convert from Future (lifted into IO eagerly via `now`, or delayed via `point`)
+   * to `IO` for some desired error type `E`
+   */
+  final def fromFuture[E, A](io: IO[Void, Future[A]])(implicit ec: ExecutionContext): IO[Throwable, A] =
+    io.attempt[Throwable].flatMap { f =>
+      IO.async { cb =>
+        f.map(
+          _.onComplete(
+            t =>
+              cb(
+                t.fold(ExitResult.Failed.apply, ExitResult.Completed.apply)
+            )
+          )
+        )
+      }
+    }
```

Comment:
in discussions with @edmundnoble, I think we both agreed that 99% of use cases for `fromFuture` will be `IO.now[Void, Future[A]`, so we start from that signature. There is no terminal case here for `IO`, so the `ExitStatus` is not handled.